### PR TITLE
[Server] Setup jsdoc and documented controller/manifest

### PR DESCRIPTION
**Pull request by BourgoisMickael**

### CODE REVIEW

File: server/src/controller/manifest.js [hunk 1]
```diff
@@ -1,8 +1,21 @@
...
+ * @public
+ * @class
+ */
 class ManifestController {
   constructor (app) {
     this.app = app
   }
 
+  /**
+   * Verify if a manifest exists
```
Comment (Arignir) [inline server/src/controller/manifest.js:13]:
```suggestion
   * Tests if a manifest exists
```

File: server/src/controller/manifest.js [hunk 2]
```diff
@@ -11,6 +24,14 @@ class ManifestController {
     return count === '1'
   }
 
+  /**
+   * Retrieve a manifest by id
```
Comment (Arignir) [inline server/src/controller/manifest.js:28]:
```suggestion
   * Retrieve a manifest given its id
```

File: server/src/controller/manifest.js [hunk 3]
```diff
@@ -23,6 +44,14 @@ class ManifestController {
     return manifestModel
   }
 
+  /**
+   * Create a manifests and a manifestContent
```
Comment (Arignir) [inline server/src/controller/manifest.js:48]:
```suggestion
   * Create a manifests filled with the given content
```

File: server/src/controller/manifest.js [hunk 4]
```diff
@@ -52,6 +90,15 @@ class ManifestController {
       .get('attributes')
   }
 
+  /**
+   * Update a manifest: create a new content and update the last_update Date
```
Comment (Arignir) [inline server/src/controller/manifest.js:94]:
```suggestion
   * Update a manifest: create a new content and update the manifest
```

File: server/src/controller/manifest.js [hunk 5]
```diff
@@ -52,6 +90,15 @@ class ManifestController {
...
+   *
+   * @public
+   * @param  {Integer}  id      Id of the manifest
+   * @param  {String}   content Content of the manifest
```
Comment (Arignir) [inline server/src/controller/manifest.js]: The two descriptions  shouldn't be identical if their type aren't the same

File: server/src/controller/manifest.js [hunk 6]
```diff
@@ -23,6 +44,14 @@ class ManifestController {
     return manifestModel
   }
 
+  /**
+   * Create a manifests and a manifestContent
+   *
+   * @public
+   * @param  {String}  name    Name of the namfest
```
Comment (person-m) [inline server/src/controller/manifest.js:51]:
```suggestion
   * @param  {String}  name    Name of the manifest
```

File: server/src/controller/manifest.js [hunk 7]
```diff
@@ -52,6 +90,15 @@ class ManifestController {
       .get('attributes')
   }
 
+  /**
+   * Update a manifest: create a new content and update the last_update Date
+   *
+   * @public
+   * @param  {Integer}  id      Id of the manifest
+   * @param  {String}   content Content of the manifest
+   * @return {Object}           Content of the manfiest
```
Comment (person-m) [inline server/src/controller/manifest.js]: This function return a object with the manifest or just the manifest content?

File: server/src/controller/manifest.js [hunk 8]
```diff
@@ -52,6 +90,15 @@ class ManifestController {
...
+   * @throws {NotFound}         If the manifest doesn't exists
```
Comment (person-m) [inline server/src/controller/manifest.js:100]:
```suggestion
   * @throws {NotFound}         If the manifest doesn't exist
```

File: server/src/controller/manifest.js [hunk 9]
```diff
@@ -65,12 +112,32 @@ class ManifestController {
     return manifestContent
   }
 
+  /**
+   * Retrieve a manifest
+   *
+   * @public
+   * @param   {Integer}  id Id of the manifest
+   * @return  {Object}      The manifest
+   * @throws  {NotFound}    If the manifest doesn't exists
```
Comment (person-m) [inline server/src/controller/manifest.js:121]:
```suggestion
   * @throws  {NotFound}    If the manifest doesn't exist
```

File: server/src/controller/manifest.js (hunk 7 above)
Comment (BourgoisMickael) [reply server/src/controller/manifest.js]: Just the manifest content

**Review by person-m — DISMISSED** [review]

**Review by Arignir — CHANGES_REQUESTED** [review]

**Review by Arignir — APPROVED** [review]

**Pull request merged**